lib/site_prism/section.rb

```diff
@@ -11,6 +11,30 @@ class Section
 
     attr_reader :root_element, :parent
 
+    class << self
```

Comment:
This area will probably need some explaining to me, as I'm not quite sure on the moving parts. Would you be willing to jump on slack?

Reply:
yes, sure. Give me a link to it pls

Reply:
Happy for this and the below (Although not sure why you can't just use class methods). (Not happy with the memoized default_search_args (2 methods below), but mentioned on slack

Reply:
It's actually the same as using class methods. I can change it to `def self.` if you prefer this

Reply:
If it doesn't facilitate a big rewrite could you (#newbie)